### DISCUSSION TITLE
How to properly use multiple data sources?

### DISCUSSION THREAD

**Discussion by vl-dud** [lululxvi/deepxde#1435] [Q&A] [answered]
Suppose I want to use PDEs and some datasets for DeepONet training - i.e. an instance of deepxde.data.PDEOperator and several instances of deepxde.data.Triple. For each data source I'm going to define a proper loss. What is the right way to do this?

**Comment by lululxvi** [comment]
I don't understand your question.

**Reply by vl-dud** [reply]
Suppose we have three data objects:
```
data_1 = dde.data.Triple(
    X_train=X_train_1,
    y_train=y_train_1,
    X_test=X_test_1,
    y_test=y_test_1
)

data_2 = dde.data.Triple(
    X_train=X_train_2,
    y_train=y_train_2,
    X_test=X_test_2,
    y_test=y_test_2
)

data_3 = dde.data.PDEOperator(
    dde.data.TimePDE(
        geomtime,
        pde,
        [],
        num_domain=1000,
        num_boundary=0,
        num_initial=0
    ),
    space,
    evaluation_points,
    10,
    num_test=200
)
```
with corresponding loss functions `loss_1`, `loss_2`, `loss_3`.
How to train a model on all objects simultaneously? The following straightforward method comes to my mind:
```
model = dde.Model(data_1, net)
stopping_condition = False
while not stopping_condition:
    for data, loss in zip([data_1, data_2, data_3], [loss_1, loss_2, loss_3]):
        model.data = data
        model.compile("adam", lr=0.001, loss=loss)
        model.train(iterations=5000)
        # check some stopping condition
        if ... :
	        stopping_condition = True
        if stopping_condition:
	        break
```			
Maybe there are other options? It is probably more correct to consider examples of all objects at each train iteration.

**Reply by lululxvi** [reply, marked as answer]
I have never done this before. Not exactly sure if your code would work. The easy way is just training first one, save the model, and then loading the model for training 2nd one, etc.

**Reply by vl-dud** [reply]
I understand you, I'll test it myself. Thanks for the answer.
